src/DevBetterWeb.Core/Services/MemberSubscriptionCreationService.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Threading.Tasks;
+using DevBetterWeb.Core.Entities;
+using DevBetterWeb.Core.Exceptions;
+using DevBetterWeb.Core.Interfaces;
+using DevBetterWeb.Core.ValueObjects;
+
+namespace DevBetterWeb.Core.Services
+{
+  public class MemberSubscriptionCreationService : IMemberSubscriptionCreationService
+  {
+    IRepository<Member> _memberRepository;
+
+    public MemberSubscriptionCreationService(IRepository<Member> repository)
+    {
+      _memberRepository = repository;
+    }
+
+    public async Task CreateSubscriptionForMemberAsync(int memberId, DateTimeRange subscriptionDateTimeRange)
+    {
+      var subscription = new Subscription();
+      subscription.MemberId = memberId;
+      subscription.Dates = subscriptionDateTimeRange;
+
+      var member = await _memberRepository.GetByIdAsync(memberId);
+      if (member is null) throw new MemberNotFoundException(memberId);
+      member.AddSubscription(subscription);
```

Comment:
That method is moved to Member.AddSubscription now. 

And yes, I'd agree that Subscription's constructor could use some refactoring, but I'll leave that for another time so that these other refactors can be finished.